### PR TITLE
Add Referring friend page for #18

### PR DESCRIPTION
**Pull request by TedGu**
### What this PR does (required):
-Added file ReferFriend.tsx for invitation dialogue
-Changed Primary Color for the theme
-Changed .eslintrc for file formatting.

### Screenshots / Videos (required):
image

### Any information needed to test this feature (required):
- sign up or log in as a new user and want to connect to another one.

### CODE REVIEW

File: client/src/pages/ReferFriend/ReferFriend.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      margin: 0,
+      padding: theme.spacing(2),
+    },
+    closeButton: {
+      position: 'absolute',
+      right: theme.spacing(1),
+      top: theme.spacing(1),
+      color: theme.palette.grey[300],
+    },
+  });
```
Comment (sina-jamshidi) [inline client/src/pages/ReferFriend/ReferFriend.tsx]: let's move these out to a separate style file and import with `useStyles` to be consistent with the rest of the project

File: client/src/pages/ReferFriend/ReferFriend.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    console.log('send invitation');
+    return;
+  };
+
+  return (
+    <Grid container spacing={0} direction="column">
+      <Dialog onClose={handleClose} aria-labelledby="invite friends" open={open} className={classes.dialogue}>
+        <DialogTitle id="invite-friends-title" onClose={handleClose}>
+          {''}
+        </DialogTitle>
```
Comment (sina-jamshidi) [inline client/src/pages/ReferFriend/ReferFriend.tsx:63]: let's see if we can change this to 
```javascript
<DialogTitle id="invite-friends-title" onClose={handleClose} />
```
So we don't have to have `{''}` as the title. That's a bit hacky and also probably not great for accessibility.

File: client/src/pages/ReferFriend/ReferFriend.tsx [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+import { Grid } from '@material-ui/core';
+import useStyles, { styles } from './useStyles';
+
+export interface DialogTitleProps extends WithStyles<typeof styles> {
+  id: string;
+  children: React.ReactNode;
+  onClose: () => void;
+}
+
+const DialogTitle = withStyles(styles)((props: DialogTitleProps) => {
```
Comment (sina-jamshidi) [inline client/src/pages/ReferFriend/ReferFriend.tsx:20]: Let's use `useStyles` to be more consistent with the other components

File: client/src/pages/ReferFriend/ReferFriend.tsx [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    <MuiDialogTitle disableTypography className={classes.root} {...other}>
+      <Typography variant="h6">{children}</Typography>
+      <IconButton aria-label="close" className={classes.closeButton} onClick={onClose}>
+        <CloseIcon />
+      </IconButton>
+    </MuiDialogTitle>
+  );
+});
+
+const DialogContent = withStyles(styles)(MuiDialogContent);
```
Comment (sina-jamshidi) [inline client/src/pages/ReferFriend/ReferFriend.tsx:32]: Let's use `useStyles` to be more consistent with the other components

**Review by sina-jamshidi — APPROVED** [review]
Looks great! If possible let's try to get a little more space between the field labels and the textfields themselves.
Once comments are addressed you can merge!

File: client/src/pages/ReferFriend/ReferFriend.tsx [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+
+  return (
+    <Grid container spacing={0} direction="column" alignItems="center" justify="center">
+      <Dialog
+        onClose={handleClose}
+        aria-labelledby="customized-dialog-title"
+        open={open}
+        style={{ width: '700px', height: '600px' }}
+      >
+        <DialogTitle id="customized-dialog-title" onClose={handleClose}>
```
[the remark this answers is not in the document]
Comment (TedGu) [reply client/src/pages/ReferFriend/ReferFriend.tsx]: delete High-Order Components API and just used hook API

File: client/src/pages/ReferFriend/ReferFriend.tsx [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+  children: React.ReactNode;
+  onClose: () => void;
+}
+
+const DialogTitle = withStyles(styles)((props: DialogTitleProps) => {
+  const { children, classes, onClose, ...other } = props;
+  return (
+    <MuiDialogTitle disableTypography className={classes.root} {...other}>
+      <Typography variant="h1">{children}</Typography>
+      {onClose ? (
```
[the remark this answers is not in the document]
Comment (TedGu) [reply client/src/pages/ReferFriend/ReferFriend.tsx]: delete

File: client/src/pages/ReferFriend/ReferFriend.tsx [hunk 7]
```diff
@@ -0,0 +1,145 @@
+import React from 'react';
```
[the remark this answers is not in the document]
Comment (TedGu) [reply client/src/pages/ReferFriend/ReferFriend.tsx:1]: fixed this bug

File: client/src/pages/ReferFriend/ReferFriend.tsx (hunk 1 above)
Comment (TedGu) [reply client/src/pages/ReferFriend/ReferFriend.tsx]: use useStyles

File: client/src/pages/ReferFriend/ReferFriend.tsx (hunk 2 above)
Comment (TedGu) [reply client/src/pages/ReferFriend/ReferFriend.tsx:63]: delete HOC

File: client/src/pages/ReferFriend/ReferFriend.tsx (hunk 3 above)
Comment (TedGu) [reply client/src/pages/ReferFriend/ReferFriend.tsx:20]: use useStyles

File: client/src/pages/ReferFriend/ReferFriend.tsx (hunk 4 above)
Comment (TedGu) [reply client/src/pages/ReferFriend/ReferFriend.tsx:32]: use useStyles

**Pull request merged**